File: client-app/app/src/main/java/com/uit/eousx/domain/model/Movie.kt
package com.uit.eousx.domain.model

data class Movie(
    val id: String,
    val title: String,
    val overview: String,
    val posterUrl: String?,
    val backdropUrl: String?,
    val releaseDate: String = "",
    val voteAverage: Double,
    val runtime: Int?,
    val genres: List<String>,
    val ageRating: String?,
    val trailerKey: String?,
    val originalTitle: String? = null
)

File: admin-web/src/types/movie.ts
export type MovieStatus = "NOW_SHOWING" | "UPCOMING";

export type Movie = {
  id: string;
  title: string;
  originalTitle?: string | null;
  overview: string;
  posterUrl?: string | null;
  backdropUrl?: string | null;
  trailerKey?: string | null;
  runtime: number;
  genres: string[];
  rating: number;
  ageRating: string;
  status: MovieStatus;
  createdAt?: string;
  updatedAt?: string;
};

export type MoviePayload = {
  title: string;
  originalTitle?: string;
  overview: string;
  posterUrl?: string;
  backdropUrl?: string;
  trailerKey?: string;
  runtime: number;
  genres: string[];
  rating: number;
  ageRating: string;
  status: MovieStatus;
};